src/adapter/src/optimize/materialized_view.rs

```diff
@@ -168,7 +168,7 @@ impl Optimize<HirRelationExpr> for Optimizer {
         trace_plan!(at: "raw", &expr);
 
         // HIR ⇒ MIR lowering and decorrelation
-        let expr = expr.lower(&self.config)?;
```

Comment:
The change might be less noisy if the metrics become part of config, or we introduce a (dreadful?) context object that contains both config and metrics?

Reply:
But anyhow it would need to touch every line that the current diff touches.

Reply:
Since we are a bit tight on time for the release cut, I'd prefer keeping it as it is.

Reply:
Ah, I commented the same thing below! Fine to do this as a follow-up if we need to rush. But I do think having the meta data encapsulated in a type would also be good for the readability of this code, not just of the diff.

Reply:
Ok, then I'll do this in a follow-up PR right away.